src/test/java/uk/gov/pay/ledger/transaction/dao/TransactionDaoSearchIT.java

```diff
@@ -508,4 +509,321 @@ public void shouldNotFilterByGatewayAccountIdWhenNotSpecified() {
+
+        Long total = transactionDao.getTotalForSearchTransactionAndParent(searchParams);
+        assertThat(total, is(2L));
+    }
+
+    @Test
+    public void searchTransactionsAndParent_shouldSearchByCardholderNameOnTransactionAndParentTransaction() {
+        transactionFixture = aTransactionFixture()
+                .withCardholderName("Mr Test")
+                .insert(rule.getJdbi());
+        TransactionFixture transactionFixtureChild = insertChildTransaction(transactionFixture,
+                now(ZoneOffset.UTC).plusDays(1));
+
+        searchParams.setAccountId(transactionFixture.getGatewayAccountId());
+        searchParams.setCardHolderName("mr");
+        List<TransactionEntity> transactionList = transactionDao.searchTransactionsAndParent(searchParams);
+
+        assertTransactionEquals(transactionList.get(0), transactionFixtureChild);
+        assertTransactionEquals(transactionList.get(0).getParentTransactionEntity(), transactionFixture);
+        assertTransactionEquals(transactionList.get(1), transactionFixture);
+
+        Long total = transactionDao.getTotalForSearchTransactionAndParent(searchParams);
+        assertThat(total, is(2L));
+    }
+
+    @Test
+    public void searchTransactionsAndParent_shouldSearchByLastDigitsCardNumberNameOnTransactionAndParentTransaction() {
+        transactionFixture = aTransactionFixture()
+                .withLastDigitsCardNumber("4242")
+                .insert(rule.getJdbi());
+        TransactionFixture transactionFixtureChild = insertChildTransaction(transactionFixture,
+                now(ZoneOffset.UTC).plusDays(1));
+
+        searchParams.setAccountId(transactionFixture.getGatewayAccountId());
+        searchParams.setLastDigitsCardNumber("4242");
+        List<TransactionEntity> transactionList = transactionDao.searchTransactionsAndParent(searchParams);
+
+        assertTransactionEquals(transactionList.get(0), transactionFixtureChild);
+        assertTransactionEquals(transactionList.get(0).getParentTransactionEntity(), transactionFixture);
+        assertTransactionEquals(transactionList.get(1), transactionFixture);
+
+        Long total = transactionDao.getTotalForSearchTransactionAndParent(searchParams);
+        assertThat(total, is(2L));
+    }
+
+    @Test
+    public void searchTransactionsAndParent_shouldSearchByCardBrandOnTransactionAndParentTransaction() {
+        transactionFixture = aTransactionFixture()
+                .withCardBrand("visa")
+                .insert(rule.getJdbi());
+        TransactionFixture transactionFixtureChild = insertChildTransaction(transactionFixture,
+                now(ZoneOffset.UTC).plusDays(2));
+
+        searchParams.setAccountId(transactionFixture.getGatewayAccountId());
+        searchParams.setCardBrands(new CommaDelimitedSetParameter("visa"));
+
+        List<TransactionEntity> transactionList = transactionDao.searchTransactionsAndParent(searchParams);
```

Comment:
[minor] It would be good to assert the count of `transactionList` as well.
(applies to all the other tests too).